app/modules/ph/endpoints_ph.py

```diff
@@ -0,0 +1,263 @@
+    paper_id: str,
+    pdf: UploadFile = File(...),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    request_id: str = Depends(get_request_id),
+    db: AsyncSession = Depends(get_db),
+):
+    paper = await cruds_ph.get_paper_by_id(db=db, paper_id=paper_id)
+    if paper is None:
+        raise HTTPException(
+            status_code=404,
+            detail="The paper does not exist.",
+        )
+
+    await save_file_as_data(
+        upload_file=pdf,
+        directory="ph/pdf",
+        filename=str(paper_id),
+        request_id=request_id,
+        max_file_size=10 * 1024 * 1024,  # 10 MB
+        accepted_content_types=["application/pdf"],
+    )
+
+    await save_pdf_first_page_as_image(
+        input_pdf_directory="ph/pdf",
+        output_image_directory="ph/cover",
+        filename=str(paper_id),
+        default_pdf_path="assets/pdf/default_ph.pdf",
+        request_id=request_id,
+        jpg_quality=95,
+    )
+
+
+@module.router.get(
+    "/ph/{paper_id}/cover",
+    status_code=200,
+)
+async def get_cover(
+    paper_id: str,
+    user: models_core.CoreUser = Depends(is_user_a_member),
+    db: AsyncSession = Depends(get_db),
+):
+    paper = await cruds_ph.get_paper_by_id(db=db, paper_id=paper_id)
+    if paper is None:
+        raise HTTPException(
+            status_code=404,
+            detail="The paper does not exist.",
+        )
+
+    return get_file_from_data(
+        default_asset="assets/images/default_cover.jpg",
+        directory="ph/cover",
+        filename=str(paper_id),
+    )
+
+
+@module.router.patch(
+    "/ph/{paper_id}",
+    status_code=204,
+)
+async def update_paper(
+    paper_id: str,
+    paper_update: schemas_ph.PaperUpdate,
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    db: AsyncSession = Depends(get_db),
+):
+    paper = await cruds_ph.get_paper_by_id(paper_id=paper_id, db=db)
+    if not paper:
+        raise HTTPException(
+            status_code=404,
+            detail="Invalid paper_id",
+        )
+
+    await cruds_ph.update_paper(
+        paper_id=paper_id,
+        paper_update=paper_update,
+        db=db,
+    )
+
+
+@module.router.delete(
+    "/ph/{paper_id}",
+    status_code=204,
+)
+async def delete_paper(
+    paper_id: str,
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.ph)),
+    db: AsyncSession = Depends(get_db),
+):
+    paper = await cruds_ph.get_paper_by_id(paper_id=paper_id, db=db)
+    if not paper:
+        raise HTTPException(
+            status_code=404,
+            detail="Invalid paper_id",
+        )
+
+    if Path.exists(Path(f"data/ph/pdf/{paper_id}.pdf")):
+        delete_file_from_data(
+            directory="ph/pdf",
+            filename=str(paper_id),
+        )
+    if Path.exists(Path(f"data/ph/cover/{paper_id}.jpg")):
+        delete_file_from_data(
+            directory="ph/cover",
+            filename=str(paper_id),
+        )
```

Comment:
I think you can call the method without checking if the file exists, `delete_file_from_data` should do that for you.

Reply:
I did that because you said : "We may want to delete a PH that does not have a PDF but use the default PDF"